### ISSUE TITLE
Using pkexec could fail if using ssh

### ISSUE THREAD

**Issue by Wang-Yue** [mesonbuild/meson#4783]
meson default to use pkexec if it's available in the system. However, pkexec needs two windows (see https://github.com/NixOS/nixpkgs/issues/18012#issuecomment-335350903) when running remotely via ssh. So it becomes very cumbersome to install a package in an ssh session. 

Is it possible for the user to choose the privilege escalation (pkexec/sudo) method? Or use sudo whenever user is away from X.

**Comment by eli-schwartz** [comment]
#11366

**Comment by xclaesse** [comment]
When I use pkexec over ssh, it asks for password in the terminal:
```
$ pkexec true
==== AUTHENTICATING FOR org.freedesktop.policykit.exec ===
Authentication is needed to run `/bin/true' as the super user
Authenticating as: root
Password: 
```

Why is it different on your system or in Meson?

**Comment by eli-schwartz** [comment]
I think it requires a logind session, because it sure doesn't work for me over ssh. I successfully type in my password, then it complains about session cookies.

I have literally never had a problem with sudo over ssh. Polkit unpredictably fails unless I obtain physical access to the device and log into a desktop. Then ssh works. Go figure.

To add to the indignity, it used to complain I'm not authorized anyway, and I had to manually configure polkit to allow the wheel user to obtain privileges. I did so purely as an experiment to see if it could work, my system didn't preconfigure this at all.
